packages/twenty-front/src/modules/auth/sign-in-up/components/SignInUpForm.tsx

```diff
@@ -127,34 +140,150 @@ export const SignInUpForm = () => {
+        {/*  Icon={() => <IconKey size={theme.icon.size.md} />}*/}
+        {/*  title={*/}
+        {/*    signInUpStep === SignInUpStep.EmailSSO*/}
+        {/*      ? 'Continue with email'*/}
+        {/*      : 'Single sign-on (SSO)'*/}
+        {/*  }*/}
+        {/*  onClick={toggleSSOMode}*/}
+        {/*  fullWidth*/}
+        {/*/>*/}
+        <HorizontalSeparator visible={true} />
 
-        {authProviders.password && (
-          <StyledForm
-            onSubmit={(event) => {
-              event.preventDefault();
-            }}
-          >
-            {signInUpStep !== SignInUpStep.Init && (
+        {authProviders.password &&
+          (signInUpStep === SignInUpStep.Password ||
+            signInUpStep === SignInUpStep.Email ||
+            signInUpStep === SignInUpStep.Init) && (
+            <StyledForm
+              onSubmit={(event) => {
+                event.preventDefault();
+              }}
+            >
+              {signInUpStep !== SignInUpStep.Init && (
+                <StyledFullWidthMotionDiv
+                  initial={{ opacity: 0, height: 0 }}
+                  animate={{ opacity: 1, height: 'auto' }}
+                  transition={{
+                    type: 'spring',
+                    stiffness: 800,
+                    damping: 35,
+                  }}
+                >
+                  <Controller
+                    name="email"
+                    control={form.control}
+                    render={({
+                      field: { onChange, onBlur, value },
+                      fieldState: { error },
+                    }) => (
+                      <StyledInputContainer>
+                        <TextInput
+                          autoFocus
+                          value={value}
+                          placeholder="Email"
+                          onBlur={onBlur}
+                          onChange={(value: string) => {
+                            onChange(value);
+                            if (signInUpStep === SignInUpStep.Password) {
+                              continueWithEmail();
+                            }
+                          }}
+                          error={showErrors ? error?.message : undefined}
+                          fullWidth
+                          disableHotkeys
+                          onKeyDown={handleKeyDown}
+                        />
+                      </StyledInputContainer>
+                    )}
+                  />
+                </StyledFullWidthMotionDiv>
+              )}
+              {signInUpStep === SignInUpStep.Password && (
+                <StyledFullWidthMotionDiv
+                  initial={{ opacity: 0, height: 0 }}
+                  animate={{ opacity: 1, height: 'auto' }}
+                  transition={{
+                    type: 'spring',
+                    stiffness: 800,
+                    damping: 35,
+                  }}
+                >
+                  <Controller
```

Comment:
this file is a bit long now. Can we split SignInUpForm and useSignInUpForm so it scales way with time? Not sure if there is an easy way